plugins/woocommerce-blocks/assets/js/atomic/utils/blocks-registration-manager/template-change-detector.ts

```diff
@@ -81,6 +82,20 @@ export class TemplateChangeDetector implements TemplateChangeDetectorSubject {
 		return templateId?.split( '//' )[ 1 ];
 	}
 
+	private getCurrentTemplateIdFromUrl( url: string ): string | undefined {
+		const path = getPath( url );
+		const isTemplatePage = path
+			? path.includes( 'site-editor.php' )
+			: false;
+		let templateId;
+
+		if ( isTemplatePage ) {
+			templateId = getQueryArg( url, 'postId' );
+		}
+
+		return templateId as string;
```

Comment:
This cast isn't necessary because we define the return type

Reply:
Unfortunately this is required because the `getQueryArg` method from the `@wordpress/url` package returns a value of type `QueryArgParsed` which is not compatible with the string type. I could change the return type of this function to be of type `QueryArgParsed` but I would have to cast it to string on the `parseTemplateId` method 😅.

Reply:
Ah ok, gotcha!